pkg/operator/clustermembercontroller/clustermembercontroller.go

```diff
@@ -83,93 +76,93 @@ func (c *ClusterMemberController) sync(ctx context.Context, syncCtx factory.Sync
 	return updateErr
 }
 
+func (c *ClusterMemberController) ensureEtcdLearnerPromotion(ctx context.Context, recorder events.Recorder) error {
+	members, err := c.etcdClient.MemberList(ctx)
+	if err != nil {
+		return fmt.Errorf("could not get etcd member list: %v", err)
+	}
+
+	var errs []error
+	for _, member := range members {
+		// Promote learner to voting member if log is in sync with leader.
+		if member.IsLearner {
+			err := c.etcdClient.MemberPromote(ctx, member.ID)
```

Comment:
Could we add a debug log level here something like:
` 		klog.V(4).Infof("attempting to promote learner with ID %d", member.ID)`